File: src/main/org/nlogo/prim/etc/_dumpextensionprims.java
package org.nlogo.prim.etc;

import org.nlogo.nvm.Reporter;
import org.nlogo.api.Syntax;

/**
 * Returns a string representation of the contents of the stack
 */
public final strictfp class _dumpextensionprims
    extends Reporter {
  @Override
  public Object report(final org.nlogo.nvm.Context context) {
    return workspace.getExtensionManager().dumpExtensionPrimitives();
  }

  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.StringType());
  }
}

File: src/main/org/nlogo/prim/etc/_dumpextensionprims.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

class _dumpextensionprims extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.StringType)
  override def report(context: Context) =
    workspace.getExtensionManager.dumpExtensionPrimitives
}
